### PR TITLE
DECISION-26436_Upgrades_angular_jasmine_karma_components

### PR DESCRIPTION
**Pull request by sundeepsapiens**
Upgrades_angular_jasmine_karma_components

JIRA LINK: https://jira.sapiens.com/browse/DECISION-26436

Angular: 7.3.0 -> 12.2.7
Jasmine: 3.3.0 -> 3.9.0
Karma: 3.1.1 -> 6.3.4
Typescript: 3.2.4 -> 4.3.5

### CODE REVIEW

File: src/lib/src/component/field.component.ts [hunk 1]
```diff
@@ -58,7 +58,7 @@ export class FieldComponent {
     return this.formModel && this.formModel.value;
   }
 
-  private get required() {
```
Comment (swatantrajain-sapiens) [inline src/lib/src/component/field.component.ts:61]: was this causing an issue?

File: src/lib/src/component/field.component.spec.ts [hunk 2]
```diff
@@ -91,15 +92,15 @@ describe('FieldComponent', () => {
     }));
 
     it('should add a label style', async(() => {
-      expect(fixture.debugElement.query(By.css('.fx-field__label')).properties['color']).toBe(this.styleObj);
```
Comment (swatantrajain-sapiens) [inline src/lib/src/component/field.component.spec.ts:94]: if this.styleObj is not being used anymore, can you remove it?

Comment (sundeepsapiens) [reply src/lib/src/component/field.component.spec.ts:94]: removed newly introduced variable labelColor and updated the specs with styleObj itself

File: src/lib/src/component/field.component.ts (hunk 1 above)
Comment (sundeepsapiens) [reply src/lib/src/component/field.component.ts:61]: yes is is causing an issue, on build it fails in AOT compilation. 

Kindly find below links for Reference, 
https://stackoverflow.com/questions/43141576/property-x-is-private-and-only-accessible-within-class-xyzcomponent
https://github.com/angular/angular/issues/11422

Comment (sundeepsapiens) [reply src/lib/src/component/field.component.ts:61]: yes is is causing an issue, on build it fails in AOT compilation.

Kindly find below links for Reference,
https://stackoverflow.com/questions/43141576/property-x-is-private-and-only-accessible-within-class-xyzcomponent
angular/angular/issues/11422(https://github.com/)

File: src/lib/src/directive/valid-submit.directive.spec.ts [hunk 3]
```diff
@@ -30,12 +34,12 @@ describe('ValidSubmitDirective', () => {
       declarations: [TestComponent]
     }).createComponent(TestComponent);
     instance = fixture.componentInstance;
-    validSubmit = spyOn(instance.validSubmitDirective.validSubmit, 'emit');
   });
 
-  beforeEach(async(() => fixture.detectChanges()));
+  beforeEach(waitForAsync(() => fixture.detectChanges()));
```
Comment (urielTwito) [inline src/lib/src/directive/valid-submit.directive.spec.ts:39]: why this? waitForAsync

**Review by urielTwito — COMMENTED** [review]
did you check this with our application?

File: src/lib/src/directive/valid-submit.directive.spec.ts (hunk 3 above)
Comment (sundeepsapiens) [reply src/lib/src/directive/valid-submit.directive.spec.ts]: async has been deprecated in angular 12 (replaced with waitForAsync), initial point of time it was causing some issue of specs failure for this file and not able to run specs, so replaced with waitForAsync. But now with async, specs are running fine so reverted the changes of waitForAsync to async.

**Comment by urielTwito** [comment]
did you check this with our application?

**Comment by sundeepsapiens** [comment]
> did you check this with our application?

yes, In local, I have replaced the dist folder of forms extensions with our application and changes are working fine

**Comment by urielTwito** [comment]
LGTM

**Pull request merged**